packages/puppeteer-core/src/common/BrowserConnector.ts

```diff
@@ -102,7 +101,7 @@ export async function _connectToCDPBrowser(
   } else if (browserWSEndpoint) {
     const WebSocketClass = await getWebSocketTransportClass();
     const connectionTransport: ConnectionTransport =
-      await WebSocketClass.create(browserWSEndpoint);
+      await WebSocketClass.create(browserWSEndpoint, headers);
```

Comment:
how difficult it is to provide a custom transport instead (already supported feature)? I guess there could be potentially many transport options and I am not sure we should include all of them in the connect options. Although headers is probably one of the most common options and might be worth including.

Reply:
> how difficult it is to provide a custom transport instead (already supported feature)?

It is not difficult because for this all I have to do is copy code from `NodeWebSocketTransport` - https://github.com/puppeteer/puppeteer/blob/main/packages/puppeteer-core/src/common/NodeWebSocketTransport.ts and add some `headers`. But it seems that such a need may arise not only for me. So it might be worth supporting it out of the box.

Reply:
Looks like it have been requested before, at least I found this one https://github.com/puppeteer/puppeteer/issues/7218. Do you know of any other open or closed issues?

Reply:
Could you please add a line `Closes #7218` to the end of your commit message/PR description?

Reply:
Done.